src/classes/RD2_EnablementService.cls

```diff
@@ -147,4 +147,28 @@ public with sharing class RD2_EnablementService {
 
         return true;
     }
+
+    /**
+     * @description Evaluates whether Enhanced Recurring Donation metadata has been deployed
+     * @return Boolean
+     */
+    public static Boolean isMetaDataDeployed() {
+        RD2_EnablementDelegate_CTRL.EnablementState state = new RD2_EnablementDelegate_CTRL.EnablementState();
+
+        try {
+            String jsonState = UTIL_CustomSettingsFacade.getRecurringDonationsSettings().RecurringDonations2EnablementState__c;
+
+            if (String.isNotBlank(jsonState)) {
+                state = (RD2_EnablementDelegate_CTRL.EnablementState) JSON.deserialize(jsonState, RD2_EnablementDelegate_CTRL.EnablementState.class);
+            }
+
+            if (String.isBlank(jsonState)) {
+                state.isMetaConfirmed = false;
+            }
+        } catch (Exception e) {
+            state.isMetaConfirmed = false;
+        }
+
+        return state.isMetaConfirmed == true;
+    }
```

Comment:
Slight revision to cache the state in a static var so that each subsequent call to the method does not need to retrieve and de-serialize the result.
```suggestion
    private static RD2_EnablementDelegate_CTRL.EnablementState enablementState;
    public static Boolean isMetaDataDeployed() {
        
        if (enablementState != null) {
            return enablementState.isMetaConfirmed == true;
        }

        enablementState = new RD2_EnablementDelegate_CTRL.EnablementState();
        enablementState.isMetaConfirmed = false;

        try {
            String jsonState = UTIL_CustomSettingsFacade.getRecurringDonationsSettings().RecurringDonations2EnablementState__c;

            if (String.isNotBlank(jsonState)) {
                enablementState = (RD2_EnablementDelegate_CTRL.EnablementState) JSON.deserialize(jsonState, RD2_EnablementDelegate_CTRL.EnablementState.class);
            }
        } catch (Exception e) { }

        return enablementState.isMetaConfirmed == true;
    }
```

Reply:
@force2b The only place this is called is RD2_ValidationService, and the below is how that is implemented.

```
    public static Boolean isRD2MetadataDeployed {
        get {
            if (isRD2MetadataDeployed == null) {
                isRD2MetadataDeployed = RD2_EnablementService.isMetaDataDeployed();
            }
            return isRD2MetadataDeployed;
        }
        set;
    }

So aren't we already caching?
```